### PR TITLE
Rewrite of PMS.isHeadless() so that the actual check is done only once

### PR DESCRIPTION
**Pull request by Nadahar**
On top of being called several times during init from PMS and DLNAMediaDatabase, it's called repeatedly from DLNAMediaInfo.parse(). I changed the logic so that the check is only done once during an application instance. This is not a bug, it's just for efficiency.

### CODE REVIEW

File: src/main/java/net/pms/PMS.java [hunk 1]
```diff
@@ -1086,9 +1085,7 @@ public static void main(String args[]) {
 				if (System.getProperty(NOCONSOLE) == null) {
 					System.setProperty(CONSOLE, Boolean.toString(true));
 				}
-			} else {
-				headless = false;
```
Comment (SubJunk) [inline src/main/java/net/pms/PMS.java]: @Nadahar why was this removed?

Comment (Nadahar) [inline src/main/java/net/pms/PMS.java]: @SubJunk The author of PMS.main() had made a local variable ```headless``` probably to avoid calling ```isHeadless()``` unnecessary. It was set on line 1090 based on the result of ```isHeadless``` and then read once in line 1100: 
```java
		if (!headless && displayProfileChooser) {
			ProfileChooser.display();
		}
```
 Since ```isHeadless()``` is now rewritten so that it only makes the test once, and after that just returns the previously estabilished status, I simply replaced the test at line 1100 (now 1097) to use isHeadless() instead:
```java
		if (!isHeadless() && displayProfileChooser) {
			ProfileChooser.display();
		}
```
I then deleted the variable and its remaining reference, the line above.

Comment (SubJunk) [inline src/main/java/net/pms/PMS.java]: @Nadahar cool, thanks for the explanation :)
